MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -1099,6 +1108,64 @@ private static Entity generateTransportedBAUnit(UnitGeneratorParameters params,
 
         return transportedUnits;
     }
+    
+    /**
+     * Worker function that generates a battle armor unit to attach to a unit of clan mechs
+     */
+    public static List<Entity> generateBAForNova(AtBScenario scenario, List<Entity> starUnits, String factionCode, int skill, int quality, Campaign campaign) {
+        List<Entity> transportedUnits = new ArrayList<>();
+        
+        // determine if this should be a nova
+        // if yes, then pick the fastest mech and load it up, adding the generated BA to the transport relationships.
+        
+        // non-clan forces and units that aren't stars don't become novas
+        if(!Faction.getFaction(factionCode).isClan() && (starUnits.size() != 5)) {
+            return transportedUnits;
+        }
+        
+        // logic copied from AtBScenario.addStar() to randomly determine if the given unit is actually going to be a nova
+        // adjusted from 11/8 to 8/6 (distribution of novas in newest AtB doc is a lot higher) so that players actually encounter novas
+        // whatever CBS is still gets no novas, so there
+        int roll = Compute.d6(2);
+        int novaTarget = 8;
+        if (factionCode.equals("CHH") || factionCode.equals("CSL")) {
+            novaTarget = 6;
+        } else if (factionCode.equals("CBS")) {
+            novaTarget = 13;
+        }
+        
+        if(roll < novaTarget) {
+            return transportedUnits;
+        }
+        
+        Entity actualTransport = null;
+        for(Entity transport : starUnits) {
+            if(transport instanceof Mech && transport.isOmni()) {
+                if((actualTransport == null) || (actualTransport.getWalkMP() < transport.getWalkMP())) {
+                    actualTransport = transport;
+                }
+            }
+        }
+        
+        if(actualTransport == null) {
+            return transportedUnits;
+        }
+        
+        // if we're generating a riding BA, do so now, then associate it with the designated transport
+        UnitGeneratorParameters params = new UnitGeneratorParameters();
+        params.setFaction(factionCode);
+        params.setQuality(quality);
+        params.setYear(campaign.getGameYear());
+        params.addMissionRole(MissionRole.MECHANIZED_BA);
+        params.setWeightClass(UNIT_WEIGHT_UNSPECIFIED);
+
+        Entity transportedUnit = generateTransportedBAUnit(params, IUnitGenerator.NO_WEIGHT_LIMIT, skill, campaign);
+        transportedUnit.setDeployRound(actualTransport.getDeployRound());
```

Comment:
Are we guaranteed `transportedUnit` is non-null?

Reply:
Probably not.